trunner/test_runner.py

```diff
@@ -111,10 +127,20 @@ def run_tests(self, tests: Sequence[TestOptions]):
 
             print(result, end="", flush=True)
 
-            if result.is_fail():
-                fail += 1
-            elif result.is_skip():
+            if result.is_skip():
                 skip += 1
+            else:
+                if result.is_fail():
+                    fail += 1
+                if is_github_actions():
+                    # New logfile is created after emulator reboot and when running a new test on host-generic-pc
+                    if 'host' in self.target.name or (tests[idx].should_reboot and 'emu' in self.target.name):
+                        pos = 0
+                    logs = self.target.dut.pexpect_proc.logfile_read.getvalue()[pos:]
+                    pos = pos + len(logs)
+
+                    logs = prepare_logs(logs)
+                    print(logs)
```

Comment:
Let's pack this code into a function. We should not touch the guts of the `Dut` class. Instead, the `Dut` class should have `get_logfiles` method that returns the triplet (logfile, logfile_read, logfile_write). Here's a possible implementation:
```suggestion
                    dump_logfiles()
                    if is_github_actions():
                        print_dut_logs(self.target.dut)
```

We don't have to keep track of the last position in the buffer. Instead, we could create a new `StringIO` instance before executing the test (not one for all tests). The workflow could look like this:
1. Set logfile to `StringIO`.
2. Execute the harness.
3. Print the content of the `StringIO`.
4. Repeat.

Additionaly, I'm not sure if it's good idea to get `logfile_read` as logs. I think logfile would fit better here. If the input is not echoed, then there is no input context in the logs.

Reply:
I don't think that `logfile` is better than `logfile_read`, in most of the cases sent commands are echoed and usually it's very difficult to read. Ok, maybe you are right, I will think about a different solution.